src/Model/FilterJob.re

```diff
@@ -135,35 +136,28 @@ module Make = (Config: Config) => {
     (false, newPendingWork, completed);
   };
 
+  /* Compare two ranked items. */
+  let compareItems = (a, b) => compare(a.Filter.score, b.Filter.score);
+
   let doActualWork =
       (
         {queue, shouldLower, filter, explodedFilter, _} as pendingWork: PendingWork.t,
-        {filtered, _}: CompletedWork.t,
+        {ranked, _}: CompletedWork.t,
       ) => {
     // Take out the items to process this frame
     let (items, queue) = Queue.take(Constants.itemsPerFrame, queue);
 
-    // Do a first filter pass to check if the item satisifies the regex
-    let filtered =
-      List.fold_left(
-        (acc, item) => {
-          let name = format(item, ~shouldLower);
-
-          if (Filter.fuzzyMatches(explodedFilter, name)) {
-            [item, ...acc];
-          } else {
-            acc;
-          };
-        },
-        filtered,
-        items,
-      );
+    // TODO: Understand why this is needed.
+    let filtered = items;
```

Comment:
If this isn't included, the results get a bit strange (i.e. just getting and passing back `filtered` unchanged, or setting it to `[]` etc). By strange I mean mostly correct, with a few missing bits and being generally nondeterministic. Including this means I get the same results as the CLI `fzy` (or equivalent based on score). Just need to poke through the Job code to understand why that is the case.